src/pyspiffe/workloadapi/default_workload_api_client.py

```diff
@@ -0,0 +1,138 @@
+from pyspiffe.workloadapi.workload_api_client import WorkloadApiClient
+from pyspiffe.internal.defaults import Defaults
+
+class DefaultWorkloadApiClient(WorkloadApiClient):
+    """
+    Default implementation for a SPIFFE Workload API Client.
+    """
+
+    # TODO: Not sure we need all the arguments, see newClient()
+    def __init__(self, spiffe_socket_path=None, exponential_backoff_policy=None,
+                 executor_service=None):
+
+        if spiffe_socket_path == None:
+            self.spiffe_socket_path=Defaults.spiffe_socket_path
+        else:
+            self.spiffe_socket_path=spiffe_socket_path
+
+        # TODO defaults for these, or delete them entirely
+        self.exponential_backoff_policy=exponential_backoff_policy
+        self.executor_service=executor_service
+
+
+    # TODO: Is this pythonic? Any benefit of this being a factory object vs
+    # just using the constructor?
+    @classmethod
+    def new_client(cls, spiffe_socket_path=None, exponential_backoff_policy=None,
```

Comment:
As we talked offline, I would leave out the retry-functionality-related stuff and implement it as a separate feature.